plugins/source/okta/codegen/recipes/recipes.go

```diff
@@ -0,0 +1,156 @@
+	Table *codegen.TableDefinition
+	// DataStruct that will be used to generate the cloudquery table
+	DataStruct interface{}
+	// SkipFields fields in go struct to skip when generating the table from the go struct
+	SkipFields []string
+	Template   string
+	Multiplex  string
+
+	ExtraColumns []codegen.ColumnDefinition
+	PKColumns    []string
+
+	PreResourceResolver   string
+	PostResourceResolver  string
+	Relations             []string
+	UnwrapEmbeddedStructs bool
+
+	Service string // Required
+
+	SkipServiceInTableName bool // Don't prepend service name to table name
+	SkipParentInTableName  bool // Don't prepend parent name to table name
+
+	// These are inferred with reflection but can be overridden
+	SubService string // Inferred from DataStruct name
+	TableName  string // singular Service + plural SubService
+
+	// These are auto calculated
+	ImportClient     bool   // true if the resource/column resolvers use the client package
+	Filename         string // Calculated from TableName
+	TableFuncName    string // Calculated from TableName
+	ResolverFuncName string // Calculated from TableFuncName
+
+	// Used for generating better table names
+	parent   *Resource
+	children []*Resource
+}
+
+var (
+	pluralizeClient *pluralize.Client
+	csr             *caser.Caser
+)
+
+func init() {
+	pluralizeClient = pluralize.NewClient()
+	csr = caser.New()
+}
+
+func (r *Resource) Infer() {
+	// Set defaults and/or infer fields
+	if r.Template == "" {
+		r.Template = "resource"
+	}
+
+	ds := reflect.TypeOf(r.DataStruct)
+	if ds.Kind() == reflect.Ptr {
+		ds = ds.Elem()
+	}
+
+	if r.Service == "" {
+		log.Fatalf("Service is required for %s", r.SubService)
+	}
+	if r.SubService == "" {
+		r.SubService = csr.ToSnake(pluralizeClient.Singular(ds.Name()))
+	}
+}
+
+func (r *Resource) GenerateNames() {
+	if r.TableName == "" {
+		// Table names are always in [<singular>...]<plural> format. Add everything in singular form and pluralize the last word later
+
+		const sep = "_"
+		var nParts []string
+		if !r.SkipServiceInTableName {
+			nParts = strings.Split(pluralizeClient.Singular(r.Service), sep)
+		}
+		p := r.parent
+		for !r.SkipParentInTableName && p != nil {
+			nParts = appendNoRepeat(nParts, strings.Split(pluralizeClient.Singular(p.SubService), sep)...)
+			p = p.parent
+		}
+		nParts = appendNoRepeat(nParts, strings.Split(pluralizeClient.Singular(r.SubService), sep)...)
+		nParts[len(nParts)-1] = pluralizeClient.Plural(nParts[len(nParts)-1])
+
+		if r.TableName == "" {
+			if len(nParts) == 0 {
+				log.Fatalf("Could not generate table name for %s.%s", r.Service, r.SubService)
+			}
+
+			r.TableName = strings.Join(nParts, sep)
+		}
+	}
+
+	r.Filename = csr.ToSnake(r.TableName) + ".go"
+	r.TableFuncName = csr.ToPascal(r.TableName)
+	r.ResolverFuncName = "fetch" + r.TableFuncName
+}
+
+// SetParentChildRelationships calculates and sets the parent and children fields on resources.
+func SetParentChildRelationships(resources []*Resource) error {
```

Comment:
Same here?

Reply:
With [`d0dc60d` (#5118)](https://github.com/cloudquery/cloudquery/pull/5118/commits/d0dc60d64919d3719be14bbfa672b1af5896e712) it's a bit more simplified and overridable.